cpp/src/parquet/file_reader.h

```diff
@@ -56,8 +57,12 @@ class PARQUET_EXPORT RowGroupReader {
   // column. Ownership is shared with the RowGroupReader.
   std::shared_ptr<ColumnReader> Column(int i);
 
+  std::shared_ptr<ColumnReader> ColumnWithIndex(int i,void* predicate, int64_t& min_index, int predicate_col, int64_t& row_index,Type::type type_num);
```

Comment:
What does this method do, and what do the arguments mean? What is `void* predicate`?

Reply:
predicate is the value that gets pushed down from the query engine, to be used to compared with column indices, and detect the page that could potentially contain the value of the predicate, I used void* for support for different parquet data types